### DISCUSSION TITLE
How to add true type fonts to SVG document

### DISCUSSION THREAD

**Discussion by yatish-infogain** [vvvv/SVG#1063] [Q&A] [answered]
We are using svg-net web for creating and manipulating SVG document and creating PNG/JPG files out of that.
We are using .Net core 3.1 and C# for this. and hosted on Azure as function app.
Our one of the requirements is to generate text in this SVG document and this text can have custom ( ttf ) fonts.
How can I add the fonts in the SVGDocument programmatically?

**Comment by H1Gdev** [comment]
FYI
https://github.com/svg-net/SVG/discussions/1060#discussioncomment-6015828

**Reply by yatish-infogain** [reply]
Thanks for reply.
I found this SVGFontManager earlier also but, not sure how to bind this manager with SVGDocument.
Do I have to create and pass the SVGFontManager  object to SVGDocument object or what?
If you can share some more details on this, it will be really very helpful.
Thank you.

**Reply by H1Gdev** [reply, marked as answer]
> Do I have to create and pass the SVGFontManager object to SVGDocument object or what?

It is automatically linked to `SvgDocument`.
Please add fonts to `SvgFontManager`.
```cs
SvgFontManager.PrivateFontPathList.Add(@"path\to\fontfile");  // if file.
SvgFontManager.PrivateFontDataList.Add(fontBytes);  // if binary.
SvgFontManager.LocalizedFamilyNames.Add(new string[] { "Yumin", "游明朝", }); // if needed.
```

**Reply by yatish-infogain** [reply]
Hi, 
Thanks for the reply.
One more question.,
As I am using ASP.Net core application, can I add all the font files to the SvgFontManager at beginning like in startup file as a service and use it everywhere I need it? or I have to add fonts at all the places I need, manually?
Also, please suggest me if there is any detailed documentation available for the SVG-Net library. they have given but it's not explaining the classes of methods in details just the code for it is available really struggle to find it, resulting to as questions like this forum.
I really appreciate your help on this. Thank you.

**Reply by H1Gdev** [reply]
> As I am using ASP.Net core application, can I add all the font files to the SvgFontManager at beginning like in startup file as a service and use it everywhere I need it? or I have to add fonts at all the places I need, manually?

SvgFontManager loads private fonts at the time of SvgDocument#Draw(), so I think it is possible to add all font file paths.
(I cannot tell exactly because I have no development experience in ASP.Net core application.)

> Also, please suggest me if there is any detailed documentation available for the SVG-Net library.

The following is documentation.
http://svg-net.github.io/SVG/

But if you want to know how to use this library, it is a shortcut to look at sample projects.

**Reply by yatish-infogain** [reply]
Hi,
Thank you very much for your help.
As you mentioned SVGFontManager loads all private font's automatically When SVGDocument.Draw() method called So, I just added fonts to SVGFontManager and then some more business logic after that called the SVGDocument.Draw() method and it worked on Azure app service also.
Unfortunately, "**_SVGFontManager loads all private font's automatically When SVGDocument.Draw() method called._**" this thing is not mentioned anywhere in the documentation. Also, the documentation link you shared is not detailed. Just wanted to share my feedback about documentation.
But in my case fortunately you helped me and it worked.
Thank you.

**Reply by ruonan0509** [reply]
hi i have a list of ttf files want to added to svg fontnamager. when added the file path to PrivateFontPathList, it is working correctly. however, if i read the file to stream and byte[] then added to PrivateFontDataList, the font is randomly off, may i konw antying i am doing wrong? thank you

oh this is some ttf files i included in my project
![image](https://github.com/user-attachments/assets/f599544e-f1ad-443c-b7d3-151b0830bfad)
the following code reading all the ttf for specific font family and adding to PrivateFontPathList is working perfectly

![image](https://github.com/user-attachments/assets/2d43cc4b-05e9-4cd6-9e77-f9182d57f0c0)

however, the following code reading the file to byte[] then adding to PrivateFontDataList is not working. the font looks randomly off.
![image](https://github.com/user-attachments/assets/435c2a54-e3c4-45aa-9f51-a630cd246381)
